stl/inc/xmemory

```diff
@@ -675,35 +686,36 @@ struct _Default_allocator_traits { // traits for std::allocator
     template <class _Other>
     using rebind_traits = allocator_traits<allocator<_Other>>;
 
-    _NODISCARD static __declspec(allocator) pointer allocate(_Alloc&, _CRT_GUARDOVERFLOW const size_type _Count) {
+    _NODISCARD static _CONSTEXPR20_DYNALLOC __declspec(allocator) pointer
+        allocate(_Alloc&, _CRT_GUARDOVERFLOW const size_type _Count) {
         return static_cast<pointer>(_Allocate<_New_alignof<value_type>>(_Get_size_of_n<sizeof(value_type)>(_Count)));
     }
 
-    _NODISCARD static __declspec(allocator) pointer
+    _NODISCARD static _CONSTEXPR20_DYNALLOC __declspec(allocator) pointer
         allocate(_Alloc&, _CRT_GUARDOVERFLOW const size_type _Count, const_void_pointer) {
         return static_cast<pointer>(_Allocate<_New_alignof<value_type>>(_Get_size_of_n<sizeof(value_type)>(_Count)));
     }
 
-    static void deallocate(_Alloc&, const pointer _Ptr, const size_type _Count) {
+    static _CONSTEXPR20_DYNALLOC void deallocate(_Alloc&, const pointer _Ptr, const size_type _Count) {
         // no overflow check on the following multiply; we assume _Allocate did that check
         _Deallocate<_New_alignof<value_type>>(_Ptr, sizeof(value_type) * _Count);
     }
 
     template <class _Objty, class... _Types>
-    static void construct(_Alloc&, _Objty* const _Ptr, _Types&&... _Args) {
+    static _CONSTEXPR20_DYNALLOC void construct(_Alloc&, _Objty* const _Ptr, _Types&&... _Args) {

```

Comment:
Here we need to add `_STD construct_at(_Ptr, _STD forward<_Types>(_Args)...)` during constant evaluation